src/components/Catalog.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react'
+import Movie from './subComponets/Movie'
+import SearchItem from './subComponets/searchItem'
+import './style/catalog.css'
+import { Link } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faHome, faLongArrowAltLeft } from '@fortawesome/free-solid-svg-icons'
+class Catalog extends Component {
+    constructor() {
+        super()
+        this.state = {
+            term: "",
+            filteredMovies: []
+        }
+    }
+
+    handleInput = (e) => {
+        const term = e.target.value.toLowerCase()
+        const { movies } = this.props.state
+        const filteredMovies = movies.filter(m => !m.isRented && m.title.toLowerCase().includes(term))
+        console.log(filteredMovies);
+        this.setState({
+            term: term,
+            filteredMovies: filteredMovies
+        })
+    }
+
+    render() {
+
+        const { users, movies } = this.props.state
+        const { rent, unRent } = this.props
+        const { userId } = this.props.match.match.params
+        const currentUser = users.find(u => u.id === userId)
+
+        return (
+            <div className="catalogContainer">
+                <div className="menu">
+                    <Link to="/" className="homeButton"> <FontAwesomeIcon icon={faLongArrowAltLeft} /></Link>
+                    <Link to="/" className="homeButton"> <FontAwesomeIcon icon={faHome} /></Link>
+                </div>
+                <div className="mainContainer">
+                    <div className="searchBar">
+                        <div className="autoComplete">
+                            <input type="text" placeholder="what do you feel like watch?" onChange={this.handleInput} className="searchInput" />
+                            <div className="results">
+                                {this.state.term.length > 0 && this.state.filteredMovies.map(m => <SearchItem key={m.id} movie={m} userId={userId} />)}
```

Comment:
when you are mesuring length, if the length is 0 a `0` might render on the screen while the filteredMovies is loading. You need to write it like this with `( )`
`{(this.state.term.length > 0) && this.state.filteredMovies.map(`